packages/dev/core/src/Materials/shaderMaterial.ts

```diff
@@ -234,6 +235,21 @@ export class ShaderMaterial extends PushMaterial {
         return this._options.needAlphaTesting;
     }
 
+    /**
+     * In case the depth buffer does not allow enough depth precision for your scene (might be the case in large scenes)
+     * You can try switching to logarithmic depth.
+     * @see https://doc.babylonjs.com/features/featuresDeepDive/materials/advanced/logarithmicDepthBuffer
+     */
+    public get useLogarithmicDepth(): boolean {
+        return this._useLogarithmicDepth;
+    }
+
+    public set useLogarithmicDepth(value: boolean) {
+        this._useLogarithmicDepth = value && this.getScene().getEngine().getCaps().fragmentDepthSupported;

```

Comment:
if not supported but set to true, we might want to notify the user that it could not be enabled